common/djangoapps/philu_commands/management/commands/autoscore_ondemand_course_ora.py

```diff
@@ -64,41 +64,46 @@ def handle(self, *args, **options):
-                                                        and response_submission_delta.days >= DAYS_TO_WAIT_AUTO_ASSESSMENT:
-                                                    try:
-                                                        # Status[0] is the status of assessment that are in waiting mode
-                                                        AssessmentWorkflow.objects.get(
-                                                            status=AssessmentWorkflow.STATUSES[0],
-                                                            course_id=course.id, item_id=block,
-                                                            submission_uuid=response_submissions.uuid)
-                                                        student = {'id': user.id, 'username': user.username,
-                                                                   'email': user.email,
-                                                                   'anonymous_user_id': anonymous_user.anonymous_user_id
-                                                                   }
-                                                        autoscore_ora(course.id, unicode(block), student)
-                                                    except AssessmentWorkflow.DoesNotExist:
-                                                        continue
-
-                                            except Submission.DoesNotExist:
+                                            response_submission = Submission.objects.filter(
+                                                student_item__student_id=anonymous_user.anonymous_user_id,
+                                                student_item__item_id=block).first()
+                                            if not response_submission:
                                                 continue
+                                            log.info('Response Created at: {created_date}'.format(
+                                                created_date=response_submission.created_at.date()
+                                            ))
+
+                                            response_submission_delta = today - response_submission.created_at.date()
+
+                                            # check if this chapter is 2 weeks older or not.
+                                            module_access_days = delta_days.days - (index_chapter * 7)
+
+                                            if (module_access_days >= DAYS_TO_WAIT_AUTO_ASSESSMENT and
+                                                    response_submission_delta.days >= DAYS_TO_WAIT_AUTO_ASSESSMENT):
+                                                try:
+                                                    # Status[0] is the status of assessment that are in waiting mode
+                                                    AssessmentWorkflow.objects.get(
```

Comment:
Just curious, why are we getting this object when we are not using it anywhere? or have I missed the usage?

P.S again I know this is not you change.

Reply:
What I got from the code is, 
This get is inside the try catch block. And if there would be no assessment in waiting mode. `get` throws `DoesNotExist` exception. In this case `autoscore_ora` method will not be called. But Yes, I think we can improve the readability by using `exists()`.

Reply:
I got it, Thanks for explaining and do agree with you we can improve readability.